api-spec/protobuf/ark/v1/indexer.proto

```diff
@@ -0,0 +1,222 @@
+      get: "/v1/commitmentTx/{txid}/swept"
+    };
+  }
+}
+
+message GetCommitmentTxRequest {
+  string txid = 1;
+}
+message GetCommitmentTxResponse {
+  int64 started_at = 1;
+  int64 ended_at = 2;
+  map<uint32, IndexerBatch> batches = 3;
+}
+
+message GetVtxoTreeRequest {
+  IndexerOutpoint batch_outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetVtxoTreeResponse {
+  repeated IndexerNode vtxo_tree = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetForfeitTxsRequest {
+  IndexerOutpoint batch_outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetForfeitTxsResponse {
+  repeated string txs = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetConnectorsRequest {
+  IndexerOutpoint batch_outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetConnectorsResponse {
+  repeated IndexerNode connectors = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetSpendableVtxosRequest {
+  string address = 1;
+  IndexerPageRequest page = 2;
+}
+message GetSpendableVtxosResponse {
+  repeated IndexerVtxo vtxos = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetTransactionHistoryRequest {
+  string address = 1;
+  int64 start_time = 2;
+  int64 end_time = 3;
+  IndexerPageRequest page = 4;
+}
+message GetTransactionHistoryResponse {
+  repeated IndexerTxHistoryRecord history = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetVtxoChainRequest {
+  IndexerOutpoint outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetVtxoChainResponse {
+  map<string, IndexerTransactions> graph = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetVirtualTxsRequest {
+  repeated string txids = 1;
+  IndexerPageRequest page = 2;
+}
+message GetVirtualTxsResponse {
+  repeated string txs = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetSweptCommitmentTxRequest {
+  string txid = 1;
+}
+message GetSweptCommitmentTxResponse {
+  repeated string swept_by = 1;
+}
+
+message SubscribeForAddressesRequest {
+  repeated string addresses = 1;
+}
+message SubscribeForAddressesResponse {
+  string address = 1;
+  repeated IndexerVtxo new_vtxos = 2;
+  repeated IndexerVtxo spent_vtxos = 3;
+}
+
+message IndexerTransactions {
+  repeated IndexerChain txs = 1;
+  int64 expires_at = 2;
+}
+
+message IndexerBatch {
+  uint64 total_batch_amount = 1;
+  uint64 total_forfeit_amount = 2;
+  int32 total_input_vtxos = 3;
+  int32 total_output_vtxos = 4;
+  int64 expires_at = 5;
+  bool swept = 6;
+}
+
+message IndexerOutpoint {
+  string txid = 1;
+  uint32 vout = 2;
+}
+
+message IndexerNode {
+  string txid = 1;
+  string parent_txid = 3;
+  int32 level = 4;
+  int32 level_index = 5;
+}
+
+message IndexerVtxo {
+  IndexerOutpoint outpoint = 1;
+  int64 created_at = 2;
+  int64 expires_at = 3;
+  uint64 amount = 4;
+  string script = 5;
+  bool is_leaf = 6;
+  bool is_swept = 7;
+  bool is_spent = 8;
+  string spent_by = 9;
+}
+
+message IndexerChain {
+  string txid = 1;
+  IndexerChainedTxType type = 2;
+}
+
+message IndexerTxHistoryRecord {
```

Comment:
This message confuses me quite a bit:

* Why would the key ever be a commitment tx?
* What is amount specifically here? Is it a "balance delta"?
*  Unsure how the `type` works: a transactions that spends a vtxo from `addressA` and sends some funds to someone, and change back to `addressA` would mean the type of this tx is both INDEXER_TX_TYPE_RECEIVED and INDEXER_TX_TYPE_SENT. No clue what INDEXER_TX_TYPE_SWEEP means 
* `created_at` is actually a cosigning request timestamp right?
* `confirmed_at` relates to its parent commitment tx(s)?
* `is_settled` Unsure what it means

Reply:
I would introduce a way to fetch all effected vtxos with each record, something similar to the `SubscribeForAddressesResponse` message. I guess fits with my suggestion to change `GetSpendableVtxos` to `GetVtxos`